model/PermissionsService.php

```diff
@@ -55,42 +54,66 @@ public function __construct(
         $this->eventManager = $eventManager;
     }
 
+    public function savePermissions(
+        bool $isRecursive,
+        core_kernel_classes_Class $class,
+        array $privilegesToSet
+    ): void {
+        $this->saveResourcePermissions($class, $privilegesToSet, $isRecursive);
+    }
+
     public function saveResourcePermissionsRecursive(
         core_kernel_classes_Resource $resource,
         array $privilegesToSet
-    ):void {
+    ): void {
         $this->saveResourcePermissions($resource, $privilegesToSet, true);
     }
 
     private function saveResourcePermissions(
         core_kernel_classes_Resource $resource,
-        array $privilegesToSet, bool $isRecursive
+        array $privilegesToSet,
+        bool $isRecursive
     ): void {
-        $currentPrivileges = $this->dataBaseAccess->getResourcePermissions($resource->getUri());
-        $addRemove = $this->strategy->normalizeRequest($currentPrivileges, $privilegesToSet);
+        $resourceURI = $resource->getUri();
+
+        $this->debug(
```

Comment:
I would kindly ask, if possible to remove other changes than the required ones at first. As I see we only changed the `getActions` method, but there are too many other changes in the code that are not just PSR-12 formatting, but logs, change methods order, etc.

Reply:
Done

Reply:
thanks